### DISCUSSION TITLE
/api/abp/api-definition   Return format

### DISCUSSION THREAD

**Discussion by easy999000** [abpframework/abp#20647] [Releases]
My interface prototype is
`Task<PagedResultDto<BoardingDto>> GetBoardingList(BoardingSearchDto input);`

/api/abp/api-definition   The interface definition returned is
`"GetBoardingListByInput": {
    "uniqueName": "GetBoardingListByInput",
    "name": "GetBoardingList",
    "httpMethod": "POST",
    "url": "api/services/app/Account/GetBoardingList",
    "supportedVersions": [],
    "parametersOnMethod": [
        {
            "name": "input",
            "typeAsString": "Wallet.Identity.Accounts.Dto.BoardingSearchDto, Wallet.Identity.Application.Contracts",
            "type": "Wallet.Identity.Accounts.Dto.BoardingSearchDto",
            "typeSimple": "Wallet.Identity.Accounts.Dto.BoardingSearchDto",
            "isOptional": false,
            "defaultValue": null
        }
    ],
    "parameters": [
        {
            "nameOnMethod": "input",
            "name": "input",
            "jsonName": null,
            "type": "Wallet.Identity.Accounts.Dto.BoardingSearchDto",
            "typeSimple": "Wallet.Identity.Accounts.Dto.BoardingSearchDto",
            "isOptional": false,
            "defaultValue": null,
            "constraintTypes": null,
            "bindingSourceId": "Body",
            "descriptorName": ""
        }
    ],
    "returnValue": {
        "type": "Volo.Abp.Application.Dtos.PagedResultDto<Wallet.Identity.Accounts.Dto.BoardingDto>",
        "typeSimple": "Volo.Abp.Application.Dtos.PagedResultDto<Wallet.Identity.Accounts.Dto.BoardingDto>"
    },
    "allowAnonymous": true,
    "implementFrom": "Wallet.Identity.Accounts.AccountController"
},`
Why does JSON have a "ByInput" ?

When I call the interface again, it keeps getting errors:
`Volo.Abp.AbpException: Could not found remote action for method: System.Threading.Tasks.Task`1[Volo.Abp.Application.Dtos.PagedResultDto`1[Wallet.Identity.Accounts.Dto.BoardingDto]] GetBoardingList(Wallet.Identity.Accounts.Dto.BoardingSearchDto) on the URL: http://identity/`

**Comment by bsogulcan** [comment]
You can add Async suffix to method name like GetBoardingListAsync, i don't know why this happen but it fixes in my experience
